File: Lista-15/java/11.java
abstract class Funcionario {
    public abstract double calcularSalario();
}

class FuncionarioHorista extends Funcionario {
    private int horas;
    private double valorHora;

    public FuncionarioHorista(int horas, double valorHora) {
        this.horas = horas;
        this.valorHora = valorHora;
    }

    @Override
    public double calcularSalario() {
        return horas * valorHora;
    }
}

class FuncionarioAssalariado extends Funcionario {
    private double salarioFixo;

    public FuncionarioAssalariado(double salarioFixo) {
        this.salarioFixo = salarioFixo;
    }

    @Override
    public double calcularSalario() {
        return salarioFixo;
    }

    public static void main(String[] args) {
        Funcionario horista = new FuncionarioHorista(160, 50);
        Funcionario assalariado = new FuncionarioAssalariado(5000);
        System.out.println(horista.calcularSalario());       // Saída: 8000
        System.out.println(assalariado.calcularSalario());   // Saída: 5000
    }
}

File: Lista-15/python/11.py
from abc import ABC, abstractmethod

class Funcionario(ABC):
    @abstractmethod
    def calcular_salario(self):
        pass

class FuncionarioHorista(Funcionario):
    def __init__(self, horas, valor_hora):
        self.horas = horas
        self.valor_hora = valor_hora

    def calcular_salario(self):
        return self.horas * self.valor_hora

class FuncionarioAssalariado(Funcionario):
    def __init__(self, salario_fixo):
        self.salario_fixo = salario_fixo

    def calcular_salario(self):
        return self.salario_fixo

# Exemplo de uso
horista = FuncionarioHorista(160, 50)
assalariado = FuncionarioAssalariado(5000)
print(horista.calcular_salario())      # Saída: 8000
print(assalariado.calcular_salario())  # Saída: 5000
